kyo-prelude/shared/src/main/scala/kyo/kernel/Flat.scala

```diff
@@ -66,6 +70,15 @@ private object FlatMacro:
         def isConcrete(t: TypeRepr) =
             t.typeSymbol.isClassDef
 
+        def isKyoData(t: TypeRepr): Boolean =
```

Comment:
My initial attempt was adding these as evidences in the `Flat` companion object but I had trouble to make it work well with type inference and the presence of the additional implicits hide the macro error message because it fails due to ambiguous implicits instead.